backend/src/utils/serverVersion.ts

```diff
@@ -3,36 +3,47 @@
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
+import { Optional } from '@mrdrogdrog/optional';
 import { promises as fs } from 'fs';
 import { join as joinPath } from 'path';
 
 import { ServerVersion } from '../monitoring/server-status.dto';
 
-let versionCache: ServerVersion;
+let versionCache: ServerVersion | undefined = undefined;
 
+/**
+ * Reads the HedgeDoc version from the root package.json. This is done only once per run.
+ *
+ * @return {Promise<ServerVersion>} A Promise that contains the parsed server version.
+ * @throws {Error} if the package.json couldn't be found or doesn't contain a correct version.
+ */
 export async function getServerVersionFromPackageJson(): Promise<ServerVersion> {
-  if (versionCache === undefined) {
-    const rawFileContent: string = await fs.readFile(
-      joinPath(__dirname, '../../package.json'),
-      { encoding: 'utf8' },
-    );
-    // TODO: Should this be validated in more detail?
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const packageInfo: { version: string } = JSON.parse(rawFileContent);
-    const versionParts: number[] = packageInfo.version
-      .split('.')
-      .map((x) => parseInt(x, 10));
-    const preRelease = 'dev'; // TODO: Replace this?
-    versionCache = {
-      major: versionParts[0],
-      minor: versionParts[1],
-      patch: versionParts[2],
-      preRelease: preRelease,
-      fullString: `${versionParts[0]}.${versionParts[1]}.${versionParts[2]}${
-        preRelease ? '-' + preRelease : ''
-      }`,
-    };
+  if (versionCache) {
+    return versionCache;
   }
+  const rawFileContent: string = await fs.readFile(
+    joinPath(__dirname, '../../../package.json'),
+    { encoding: 'utf8' },
+  );
+  const packageInfo = JSON.parse(rawFileContent) as { version: string };
+  const versionParts = Optional.ofNullable(packageInfo.version)
+    .orThrow(() => new Error('No version found in root package.json'))
+    .map((version) => /^(\d+).(\d+).(\d+)(?:-(\w+))?$/g.exec(version))
+    .orElseThrow(
+      () =>
+        new Error(
+          `Version from package.json is malformed. Got ${packageInfo.version}`,
+        ),
+    );
+  return {
```

Comment:
Were do you set the `versionCache`? I think this is missing here and it would parse the file each time this is called.